lib/MusicBrainz/Server/Data/Work.pm

```diff
@@ -70,38 +70,61 @@ sub find_by_artist
 
     # For the basic work query
     push @where_query, 'work.id = s.work';
-    push @where_args, ($artist_id) x 2;
+
+    # Select works that are related to recordings for this artist
+    my $performers_query = '
+        SELECT entity1 AS work
+            FROM l_recording_work
+            JOIN recording ON recording.id = entity0
+            JOIN artist_credit_name acn
+                    ON acn.artist_credit = recording.artist_credit
+        WHERE acn.artist = ?
+    ';
+
+    my $writers_query = '
+        -- Select works that this artist is related to
```

Comment:
nitpick: this might as well be a perl comment instead of sending it to the db server